omod/src/main/java/org/openmrs/module/coreapps/fragment/controller/visit/ParserEncounterIntoSimpleObjects.java

```diff
@@ -131,6 +135,14 @@ private SimpleObject parseObs(Obs obs, Locale locale) {
         else {
             SimpleObject simpleObject = SimpleObject.create("obsId", obs.getObsId());
 
+            if (obs.getConcept().isNumeric()) {
+                Boolean isIntegerValue = false;
+                if (!conceptService.getConceptNumeric(obs.getConcept().getConceptId()).isAllowDecimal()) {
+                    isIntegerValue = true;
+                }
+                simpleObject.put("isIntegerValue", isIntegerValue);
```

Comment:
How about `allowDecimal` instead of `isIntegerValue`?

Reply:
The only reason that I use `isIntegerValue` it's because this way I can evaluate the `true` instead of the false case because we want the case when we cannot use the decimals...